src/Integration.Vsix/CFamily/VcxProject/FileConfig.cs

```diff
@@ -28,8 +29,99 @@ namespace SonarLint.VisualStudio.Integration.Vsix.CFamily.VcxProject
 {
     internal class FileConfig : IFileConfig
     {
+        private static bool TryGetCompilerPathFromClCompilerPath(ILogger logger, VCConfiguration vcConfig, IFileSystem fileSystem, out string compilerPath)
+        {
+            compilerPath = vcConfig.GetEvaluatedPropertyValue("ClCompilerPath");
+            if (string.IsNullOrEmpty(compilerPath))
+            {
+                logger.WriteLine("\"ClCompilerPath\" was not found.");
+                return false;
+            }
+
+            if (!fileSystem.File.Exists(compilerPath))
+            {
+                logger.WriteLine($"Compiler path (based on \"ClCompilerPath\") \"{compilerPath}\" does not exist.");
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool TryGetCompilerPathFromExecutablePath(ILogger logger, VCConfiguration vcConfig, IFileSystem fileSystem, out string compilerPath)
+        {
+            compilerPath = default;
+            var executablePath = vcConfig.GetEvaluatedPropertyValue("ExecutablePath");
+            if (string.IsNullOrEmpty(executablePath))
+            {
+                logger.WriteLine("\"ExecutablePath\" was not found.");
+                return false;
+            }
+
+            var toolExe = vcConfig.GetEvaluatedPropertyValue("CLToolExe");
+            if (string.IsNullOrEmpty(toolExe))
+            {
+                logger.WriteLine("\"CLToolExe\" was not found, falling back to cl.exe.");
+                toolExe = "cl.exe";
```

Comment:
Now, that there is fallback here. Do you think that it makes sense to have a test for this case?